server/src/main/java/org/elasticsearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -219,7 +219,7 @@ ClusterState execute(ClusterState currentState) {
                     // We have non-dynamic settings and open indices. We will unassign all of the shards in these indices so that the new
                     // changed settings are applied when the shards are re-assigned.
                     routingTableBuilder = RoutingTable.builder(allocationService.getShardRoutingRoleStrategy(), currentRoutingTable);
-                    for (Index index : openIndices) {
+                    for (Index index : new HashSet<>(openIndices)) {
```

Comment:
This `HashSet` constructor itself iterates over its parameter. Are we concerned about modifications while the constructor is running?

Reply:
No, this is single-threaded code. The openIndices object never escapes this thread.